### ISSUE TITLE
language R: pandoc 2.0 doesn't work

### ISSUE THREAD

**Issue by ickc** [travis-ci/travis-ci#8669]
@craigcitro, @hadley, and @jimhester,

By the way, [Building an R Project - Travis CI](https://docs.travis-ci.com/user/languages/r/) askes to call all 3 people, but [travis-build/r.rb at master · travis-ci/travis-build](https://github.com/travis-ci/travis-build/blob/master/lib/travis/build/script/r.rb) only has 2 names.

If `pandoc_version` is specified as 2.0, the build will fail. I think the problem is from https://github.com/travis-ci/travis-build/blob/master/lib/travis/build/script/r.rb, which has too much things "burnt in", e.g. the pattern of the name of the file.

I have 2 examples in different languages that has better handling on it:

- Python: [pypandoc/pandoc_download.py at master · bebraw/pypandoc](https://github.com/bebraw/pypandoc/blob/master/pypandoc/pandoc_download.py#L25-L63)
- bash: [pantable/.travis.yml at master · ickc/pantable](https://github.com/ickc/pantable/blob/master/.travis.yml#L56-L69)

Unfortunately, I don't write Ruby so I can't help here. But essentially it is just using regex, keeping the filename pattern as general as possible, and capture the resultant filename / version from the HTML. (In these scripts, I allow the version to be "latest", which will uses whatever "latest" GitHub Releases there is.)

Thanks.

**Comment by BanzaiMan** [comment]
Please refrain from mentioning @hadley. He is no longer a maintainer. (We have removed his name from the logs for that reason.)

**Comment by BanzaiMan** [comment]
@leezer3 Thanks! I've updated the docs.

**Comment by jimhester** [comment]
FWIW pandoc 2.0 [works on linux builds as-is](https://travis-ci.org/jimhester/travis-test/builds/294886795#L799-L811). The place it will fail in on macOS, as the suffix was changed from -osx to -macOS.

Is there a reason you are using a regex on the HTML release page and not using the appropriate [GitHub API call](https://developer.github.com/v3/repos/releases/#get-a-release-by-tag-name) in your examples?

**Comment by ickc** [comment]
>Is there a reason you are using a regex on the HTML release page and not using the appropriate GitHub API call in your examples?

Didn’t study GitHub API enough to do that. Can it also apply to `latest` case? If so I should adapt this to the Python/bash examples too.

**Comment by jimhester** [comment]
Yes, there is a different endpoint to get the latest release https://developer.github.com/v3/repos/releases/#get-the-latest-release

**Comment by BanzaiMan** [comment]
Is there anything left to do here?
